packages/app/src/SignInPage.tsx

```diff
@@ -1,17 +1,31 @@
 import { Title } from '@mantine/core';
-import { Logo, SignInForm } from '@medplum/react';
-import React from 'react';
+import { Logo, SignInForm, useMedplum } from '@medplum/react';
+import React, { useCallback, useEffect } from 'react';
 import { useNavigate, useSearchParams } from 'react-router-dom';
 import { getConfig, isRegisterEnabled } from './config';
 
 export function SignInPage(): JSX.Element {
+  const medplum = useMedplum();
   const navigate = useNavigate();
   const [searchParams] = useSearchParams();
   const config = getConfig();
 
+  const navigateToNext = useCallback(() => {
+    // only redirect to next if it is a pathname to avoid redirecting
+    // to a maliciously crafted URL, e.g. /signin?next=https%3A%2F%2Fevil.com
+    const nextUrl = searchParams.get('next');
+    navigate(nextUrl?.startsWith('/') ? nextUrl : '/');
+  }, [searchParams, navigate]);
+
+  useEffect(() => {
+    if (medplum.getProfile()) {
```

Comment:
We should only do this if `next` is present.

The use case to keep in mind is "Add another account".  If the user is already signed-in, they can use the top-right menu and click "Add another account".  That will navigate to "/signin" to add another session.

Reply:
@codyebberson mind taking another look?